javascript/ql/src/Security/CWE-830/FunctionalityFromUntrustedSource.ql

```diff
@@ -0,0 +1,57 @@
+/**
+ * @name Inclusion of untrusted functionality by a HTML element.
+ * @description Including untrusted functionality by a HTML element
+ *              opens up for potential man-in-the-middle attacks.
+ * @kind problem
+ * @problem.severity warning
+ * @security-severity 8.1
+ * @precision high
+ * @id js/functionality-from-untrusted-source
+ * @tags security
+ *       external/cwe/cwe-830
+ */
+
+import javascript
+import semmle.javascript.HTML
+
+bindingset[host]
+predicate isAllowedHost(string host) { host.toLowerCase().regexpMatch("localhost(:[0-9]+)?/.*") }
```

Comment:
This predicate name is relative to the query semantics.
Could we instead use a name that is relative to the predicate semantics?

```suggestion
predicate isLocalhostPrefix(string host) { host.toLowerCase().regexpMatch("localhost(:[0-9]+)?/.*") }
```

Reply:
Also: the prefix 127.0.0.1 (and friends?) should be accepted here since they are even safer (does not rely on dns)